### ISSUE TITLE
Shell integration: Screen reader support: Sync prompt with xterm.js textarea

### ISSUE THREAD

**Issue by Tyriar** [microsoft/vscode#139403]
Parent issue https://github.com/microsoft/vscode/issues/133084

Blocked on:

- https://github.com/microsoft/vscode/issues/139406

Notes:

- Currently we clear the invisible text area in xterm.js when we're unsure what the prompt's state is, with real command tracking we will be able to synchronize the textarea with the prompt and cursor to get left/right/backspace/delete correctly announcing the textual changes to screen reader users
- Considerations:
  - This might need to be a new API in xterm.js so it avoids clearing the textarea

**Comment by meganrogge** [comment]
Investigated this and it seems to be mostly working fine for screen readers. Word navigation does not work at the moment and we might want to consider announcing the output
